contributor-guide/modules/ROOT/pages/testing/boost-test-matrix.adoc

```diff
@@ -1,286 +1,178 @@
+
+. To run B2's regression tests, run `python test\_all.py` from the `<boost-root>/tools/build/test` directory. 
+
+== Running Boost's Automated Regression and Reporting
+
+To run all Boost regression tests and report the results back to the Boost community:
+
+. Create a new directory for the branch you want to test.
+. Copy the https://raw.githubusercontent.com/boostorg/regression/develop/testing/src/run.py[run.py] script into that directory.
+. Run `python run.py <options> [<commands>]` with these three options, plus any other <<_run_options>> or <<_run_commands>> you wish to employ:
 
+[#footnote1-location]
+[cols="1,2",options="header",stripes=even,frame=none]
+|===
+| *Required Option* | *Description*
+| `--runner=` | Your choice of name that identifies your results in the reports link:#footnote1[(1)].
+| `--toolsets=` | One or more toolsets that you want to test with. If a `--toolsets` option is not provided, the script will try to use the platform's default toolset (for example: GCC for most Unix-based systems).
+| `--tag=` | The tag (i.e. branch) you want to test. Currently only `--tag=develop` and `--tag=master` are valid.
+|===
 
-### Providing detailed information about your environment
+For example, if you are using the runner id `Metacomm` with the GCC and MSVC compilers, on the develop branch:
 
+[source,bash]
+----
+python run.py --runner=Metacomm --toolsets=gcc-4.2.1,msvc-8.0 --tag=develop
+----
 
-Once you have your regression results displayed in the
- Boost-wide reports, you may consider providing a bit more
- information about yourself and your test environment. This
- additional information will be presented in the reports on a
- page associated with your runner ID.
+By default, the script runs in what is known as _full mode_: on each `run.py` invocation all the files that were left in place by the previous run, including the binaries for the successfully built tests and libraries, are deleted, and everything is rebuilt once again from scratch. For a quicker alternative, refer to <<_run_incrementally>>.
 
+**Note**: If you are behind a firewall/proxy server, everything should still "just work". In the rare cases when it doesn't, you can explicitly specify the proxy server parameters through the `--proxy` option, for example:
 
-By default, the page's content is just a single line coming
- from the comment.html file in your run.py
- directory, specifying the tested platform. You can put online a
- more detailed description of your environment, such as your
- hardware configuration, compiler builds, and test schedule, by
- simply altering the file's content. Also, please consider
- providing your name and email address for cases where Boost
- developers have questions specific to your particular set of
- results.
+[source,bash]
+----
+python run.py ... --proxy=http://www.someproxy.com:3128
+----
 
+=== Run Commands
 
-### Incremental runs
+The following commands can be specified for the `run.py` script: 
 
+* `cleanup`
```

Comment:
We should describe what these options do or just tell the user to run `--help` to get a complete list of options.